hubverse_annotator/utils.py

```diff
@@ -131,6 +131,31 @@ def get_reference_dates(forecast_table: pl.DataFrame) -> list[datetime.date]:
     return forecast_table.get_column("reference_date").unique().to_list()
 
 
+def get_initial_window_range(
+    forecast_table: pl.DataFrame,
+    forecast_date_col: str = "target_end_date",
+    extra_weeks: int = 6,
+) -> tuple[datetime.date, datetime.date]:
+    """
+    Determine the initial x-axis window for the most
+    recent portion of the data. The window ends at the
+    latest observed date and starts a few weeks before.
+    """
+    if forecast_table.is_empty():
+        return None, None
+    forecasts = forecast_table.with_columns(
+        pl.col(forecast_date_col).cast(pl.Datetime)
+    )
+    max_forecast_date = forecasts.select(
+        pl.col(forecast_date_col).max()
+    ).item()
+    if max_forecast is None:
+        return alt.Undefined
```

Comment:
Try to condense this. I'm having a hard time imagining how `max_forecast` could be `None` if `forecast_table` is not empty.